### ISSUE TITLE
There should be a test to verify that theme has been applied successfully. The test can simply check if light/dark classes are applied on document.

### ISSUE THREAD

**Issue by hariombalhara** [calcom/cal.com#5977]
There should be a test to verify that theme has been applied successfully. The test can simply check if light/dark classes are applied on document.

_Originally posted by @hariombalhara in https://github.com/calcom/cal.com/issues/5976#issuecomment-1346051434_

**Comment by abhisheksunil2201** [comment]
I would like to attempt adding the test to verify theme. 
/attempt #5977

**Comment by abhisheksunil2201** [comment]
@hariombalhara The theme test needs to be done only for the public booking page right? Because as I understand, the themes apply only to public booking pages currently.

**Comment by hariombalhara** [comment]
Yeah these tests are meant for that only. To be clear, we need to improve existing tests to verify theme in a better way.
e.g. https://github.com/calcom/cal.com/blob/main/packages/embeds/embed-core/playwright/tests/inline.e2e.ts#L17 this is how we verify theme right now. We just check that theme query param exist, but we don't check that the theme has been applied to the page.

**Comment by usmaann1** [comment]
/attempt #5977
Test Plan: Verification of Theme Application

Objective:
The objective of this test plan is to verify the successful application of a theme by checking if the appropriate light/dark classes are applied to the document. The test plan includes both manual tests and a list of automated tests to ensure thorough verification.

Test Scope:
The test scope includes verifying the application of the theme on the document across different components and sections.

Test Environment:
The test environment should consist of:

- The application or system where the theme is being applied.
- Access to the document or webpage where the theme should be visible.
- Test data or content that covers different components and sections of the document.

Test Plan:
4.1 Manual Tests:

4.1.1 Visual Inspection:
Objective: To visually inspect the document and validate the application of the light/dark theme.

Steps:
- Open the document or webpage in the expected environment.
- Manually review the different components, sections, and elements.
- Verify that the appropriate light/dark classes are applied according to the selected theme.
- Ensure that the applied theme is visually consistent across the document.

4.1.2 Interaction Testing:
Objective: To test the behavior of the document and interactions with the applied theme.

Steps:
- Interact with different components and sections, such as buttons, menus, input fields, etc.
- Verify that the applied theme is preserved during interactions.
- Check if the interaction elements (e.g., hover, focus) are appropriately styled according to the theme.

4.1.3 Cross-Browser Testing:
Objective: To verify that the theme is applied consistently across different browsers.
Steps:
- Open the document or webpage in various supported browsers (e.g., Chrome, Firefox, Safari, Edge).
- Perform visual inspection to ensure the applied theme is consistent across all browsers.

4.2 Automated Tests:

4.2.1 CSS Class Verification:
Objective: To automatically verify the presence of the light/dark classes in the document's HTML.
Steps:
- Utilize a testing framework (e.g., Selenium, Puppeteer) to automate the test.
- Select elements representative of different components and sections.
- Retrieve the HTML code for the selected elements.
- Verify if the appropriate light/dark classes are present in the retrieved HTML code.

4.2.2 Theme Persistence:
Objective: To ensure the theme is consistently applied during user interactions.
Steps:
- Use an automated testing framework to simulate user interactions (e.g., clicking buttons, entering text).
- Interact with different components and sections.
- Verify if the applied theme remains intact during the simulated interactions.

4.2.3 Responsive Design Testing:
Objective: To verify the theme's responsiveness on different devices and screen sizes.
Steps:
- Utilize a responsive design testing framework (e.g., BrowserStack, Responsive Checker).
- Test the document's responsiveness across various devices and screen resolutions.
- Verify that the applied theme is correctly displayed and adapts to different screen sizes.

Test Reporting:
For each test, document the observed results, any deviations from expected behavior, and any defects found during testing. Report and track the issues found using a bug-tracking system.

Test Completion Criteria:
The test execution is considered complete when:

- All manual tests have been performed and documented.
- All automated tests have been executed and their results documented.
- Any identified issues have been reported and tracked.
